src/main/java/net/sargue/time/jsptags/FormatSupport.java

```diff
@@ -13,149 +13,160 @@
+	protected String pattern;
+	/** The style attribute. */
+	protected String style;
+	/** The zoneId attribute. */
+	protected ZoneId zoneId;
+	/** The locale attribute. */
+	protected Locale locale;
+	/** The var attribute. */
+	private String var;
+	/** The scope attribute. */
+	private int scope;
+
+	/**
+	 * Constructor.
+	 */
+	public FormatSupport() {
+		super();
+		init();
+	}
+
+	private void init() {
+		var = null;
+		value = null;
+		pattern = null;
+		style = null;
+		zoneId = null;
+		locale = null;
+		scope = PageContext.PAGE_SCOPE;
+	}
+
+	/**
+	 * 
+	 * @param var the variable to store the result in
+	 */
+	public void setVar(final String var) {
+		this.var = var;
+	}
+
+	/**
+	 * 
+	 * @param scope the scope to put the variable in
+	 * @see #setVar(String)
+	 */
+	public void setScope(final String scope) {
+		this.scope = Util.getScope(scope);
+	}
+
+	/**
+	 * Formats the given instant or partial.
+	 */
+	public int doEndTag() throws JspException {
+		if (value == null) {
+			if (var != null) {
+				pageContext.removeAttribute(var, scope);
+			}
+			return EVAL_PAGE;
+		}
+
+		// Create formatter
+		DateTimeFormatter formatter;
+		if (pattern != null) {
+			formatter = DateTimeFormatter.ofPattern(pattern);
+		} else if (style != null) {
+			formatter = Util.createFormatterForStyle(style);
+		} else {
+			// use a medium date (no time) style by default; same as jstl
+			formatter = Util.createFormatterForStyle("M-");
+		}
+
+		// set formatter locale
+		Locale locale = this.locale;
+		if (locale == null) {
+			locale = Util.getFormattingLocale(pageContext, true, DateFormat.getAvailableLocales());
+		}
+		if (locale != null) {
+			formatter = formatter.withLocale(locale);
+		}
+
+		// set formatter timezone
+		ZoneId zoneId = this.zoneId;
+		if (zoneId == null) {
+			zoneId = ZoneIdSupport.getZoneId(pageContext, this);
+		}
+		if (zoneId != null) {
+			formatter = formatter.withZone(zoneId);
+		} else {
+			if (value instanceof Instant || value instanceof LocalDateTime || value instanceof OffsetDateTime
+					|| value instanceof OffsetTime || value instanceof LocalTime)
+				// these time objects may need a zone to resolve some patterns
+				// and/or styles, and as there is no zone we revert to the
+				// system default zone
+				formatter = formatter.withZone(ZoneId.systemDefault());
+		}
+
+		// format value
+		String formatted = null;
+		if (value instanceof TemporalAccessor) {
+			formatted = formatter.format((TemporalAccessor) value);
+		} else {
+			throw new JspException("value attribute of format tag must be a TemporalAccessor," + " was: "
+					+ value.getClass().getName());
+		}
+
+		if (var != null) {
+			pageContext.setAttribute(var, formatted, scope);
+		} else {
+			try {
+				pageContext.getOut().print(formatted);
+			} catch (IOException ioe) {
+				throw new JspTagException(ioe.toString(), ioe);
+			}
+		}
+
+		return EVAL_PAGE;
+	}
+
+	@Override
+	public void release() {
+		init();
+		super.release();
```

Comment:
@sargue  would you like this super.release() call removed? Given that this is overriding a parent class method it probably should be there.

Reply:
Yes please. This is code from the original library that it's supposed to be working perfectly fine, so I prefer not to modify it unless there's some specific problem.

Why the original author didn't call the parent method I don't know.

Reply:
Done